moonstreamapi/moonstreamapi/routes/subscriptions.py

```diff
@@ -559,3 +566,158 @@ async def list_subscription_types() -> data.SubscriptionTypesListResponse:
         raise MoonstreamHTTPException(status_code=500, internal_error=e)
 
     return data.SubscriptionTypesListResponse(subscription_types=results)
+
+
+@router.get(
+    "/is_contract",
+    tags=["subscriptions"],
+    response_model=data.ContractInfoResponse,
+)
+async def address_info(request: Request, address: str):
+    """
+    Looking if address is contract
+    """
+
+    user_token = request.state.token
+
+    try:
+        Web3.toChecksumAddress(address)
+    except ValueError as e:
+        raise MoonstreamHTTPException(
+            status_code=400,
+            detail=str(e),
+            internal_error=e,
+        )
+
+    try:
+        resource = bc.list_resources(
+            token=user_token,
+            params={
+                "type": "nodebalancer-access",
+            },
+        )
+    except Exception as e:
+        logger.error(f"Error reading contract info from Brood API: {str(e)}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    access_id = resource.resources[0].resource_data["access_id"]
+
+    if not access_id:
+        raise MoonstreamHTTPException(
+            status_code=404,
+            detail="Not found access id",
+        )
+
+    contract_info = {}
+
+    for blockchain_type in AvailableBlockchainType:
+        try:
+            # connnect to blockchain
+
+            futures = []
+
+            with ThreadPoolExecutor(max_workers=5) as executor:
```

Comment:
I suspect we need timeouts for all thread executors, so there will be no freeze tasks

Reply:
Add in result method.